### PR TITLE
Add logging

### PR DESCRIPTION
**Pull request by inglesp**
This PR:

* allows us to log SQL sent to the presto backend (by setting `LOG_SQL`)
* allows us to get debug logs from underlying libraries (by setting `LOG_LEVEL`)
* changes from `print()` to `logger.info()` for presenting information to users, which means that the output is now slightly noisier -- we may want to revisit this

### CODE REVIEW

**Comment by inglesp** [comment]
Thanks to @ghickman for the earlier logging seminar.

File: cohortextractor/cohortextractor.py [hunk 1]
```diff
@@ -158,7 +160,17 @@ def _generate_cohort(
     index_date_range=None,
     skip_existing=False,
 ):
-    print("Running. Please wait...")
+    logger.info(
+        "Generating cohort for {study_name} in {output_dir}",
```
Comment (ghickman) [inline cohortextractor/cohortextractor.py:164]: Missing `f` string prefix here?

Side note: stdlib logging is [optimised](https://docs.python.org/3/howto/logging.html#optimization) when using `%` style formats.  I suspect we'll never see the benefit of it here but thought you could use just a smidge more logging info for the day!

**Review by evansd — APPROVED** [review]
This looks good (though even with structlog, Python logging still manages to make me feel sad)

File: cohortextractor/cohortextractor.py (hunk 1 above)
Comment (inglesp) [reply cohortextractor/cohortextractor.py]: Well spotted!

Comment (bloodearnest) [reply cohortextractor/cohortextractor.py]: Using f strings is fine here IMO - it's clear, consistent, and negligible performance overhead is not a problem in this scenario.

FWIW, it's not %-style per se (you can use {} style if you've set a logging.Formatter with style='{') - it's the delayed string interpolation which is the optimisation.

e.g. If you do:

    logger.info('%s', 'msg')
    # or with style='{'
    logger.info('{}', 'msg')

Then the string will only be interpolated iff the log message is actually emitted by a handler. i.e. if it's a DEBUG level log, and you're only logging INFO, the interpolation will be skipped. Can be useful if you have a lot of DEBUG level logs with logs of embedded string interpolation - avoid the costs if not turned on.

**Pull request merged**